### PR TITLE
UCP/TAG-OFFLOAD: unification of rts packet for rndv/tag-offload

### PR DESCRIPTION
**Pull request by hoopoepg**
- in case if SW rndv going to be used then pack
  rts header + rkey to tag-offload message

### CODE REVIEW

**Comment by brminich** [comment]
plz, remove `ucp_tag_offload_fill_sw_rts` as well

**Comment by hoopoepg** [comment]
removed
thank you

File: src/ucp/tag/offload.c [hunk 1]
```diff
@@ -429,61 +391,35 @@ static ucs_status_t ucp_tag_offload_eager_zcopy(uct_pending_req_t *self)
...
     } else {
         rndv_hdr_len    = sizeof(ucp_sw_rndv_hdr_t);
         rndv_hdr        = ucs_alloca(rndv_hdr_len);
         rndv_hdr->flags = 0;
-    }
 
-    rndv_hdr->super.sender_uuid = ep->worker->uuid;
-    rndv_hdr->super.reqptr      = (uintptr_t)req;
-    rndv_hdr->length            = req->send.length;
+        rndv_hdr->super.sender_uuid = ep->worker->uuid;
```
Comment (brminich) [inline src/ucp/tag/offload.c]: why don't you pack the whole RTS here as well? That will not work, because you treat it as RTS on the receiver

Comment (hoopoepg) [reply src/ucp/tag/offload.c]: fixed

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5386/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5388/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5390/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5396/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:ornl:retest

**Comment by yosefe** [comment]
@brminich good to go?

**Pull request merged**